mautic/mautic
# Gated Video Embeds

| Q  | A
| --- | ---
| New feature? | Y
| Related user documentation PR URL | mautic/documentation#72

#### Description:

This PR adds the long awaited gated videos feature. Gated videos allow you to pause the video at a set time and then show a contact capture form that must be filled out in order to continue watching the video. The purpose is to increase engagement by creating additional inbound contact points.

Note: This feature requires your site (or landing pages) to have jQuery included prior to the inclusion of mtc.js.

Note: If you choose to use Vimeo as your video provider, you must also manually include the Vimeo Froogaloop javascript file in your document head. (The url to the froogaloop js is `https://f.vimeocdn.com/js/froogaloop2.min.js`.)

#### Steps to test this PR:
1. Apply the PR to your dev environment
2. Clear the cache via `php app/console cache:clear --env=dev`
3. Create a new landing page with a template, and use the Skyline template (only for testing, this feature works with on any page where the Mautic tracking javascript is embedded).
4. Place the cursor into one of the text editor areas (where you can click to see the Froala editor) and the editor where appear. To the right of the "Insert Image" button, is a new "Insert Video" button. It looks like this: <img width="634" alt="screen shot 2016-07-25 at 4 02 03 pm" src="https://cloud.githubusercontent.com/assets/718028/17115436/40fb4e64-5281-11e6-99e4-c0e689aba906.png">
5. In the modal that opens, enter a URL, gate time, and then select the form to display.  
    a. URL - The URL to the video. Supported providers are YouTube, Vimeo, and MP4  
    b. Gate time - The time (in seconds) at which you want to pause the video and display the form.  
    c. Form - Select the form which you would like to display when the video reaches the gate time.
6. Once you have the video embedded, close the builder.
7. Go to the code view of the page, and insert the jQuery script tag into the head of your document via `<script type="text/javascript" src="/app/bundles/CoreBundle/Assets/js/libraries/2.jquery.js"></script>`
8. Insert the following CSS tag as well `<link rel="stylesheet" href="/media/css/mediaelementplayer.css" />`
9. Leave the code view, save the page, then navigate to the landing page in your mautic dev environment.
10. You will see that the video is embedded. Play the video and it will pause at your selected gate time, and pull in the form that you selected as well. Here you will fill out the form, and see that the video will continue playing after the submission is processed.
